packages/dd-trace/src/plugins/util/web.js

```diff
@@ -270,15 +270,26 @@ function wrapEnd (req) {
     return returnValue
   }
 
-  Object.defineProperty(res, 'end', {
-    configurable: true,
-    get () {
-      return this._datadog_end
-    },
-    set (value) {
-      this._datadog_end = scope.bind(value, req._datadog.span)
-    }
-  })
+  res._datadog_req = req
+  res._datadog_scope = scope
+
+  if (!res._datadog_instrumented) {
+    const target = Reflect.ownKeys(res).includes('end') ? res : res.constructor.prototype
+    Object.defineProperty(target, 'end', {
+      configurable: true,
+      get () {
+        return this._datadog_end || end
+      },
+      set (value) {
+        if (this._datadog_scope && this._datadog_req && this._datadog_req._datadog) {
```

Comment:
`this._datadog_req` is repeated a few times and is never assigned to, so it would be cleaner to move it to a const.

Reply:
Sure.